dm/pkg/exstorage/utils.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exstorage
+
+import (
+	"context"
+	"net/url"
+	"path"
+	"strings"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb/br/pkg/storage"
+)
+
+// AdjustS3Path adjust s3 rawURL, add uniqueId into s3 path.
+func AdjustS3Path(rawURL string, uniqueID string) (string, error) {
+	if rawURL == "" {
```

Comment:
these logic is repeated with `IsS3Path`, maybe we can declare a `isS3Path` that returns `(bool, URL, error)` to make use of parsed URL, `AdjustS3Path` and `IsS3Path` will use `isS3Path` internally. (Also you can give it a better name)

Reply:
ok

Reply:
`AdjustS3Path` has been changed to `AdjustPath`, and it may no need to check s3, so may be `isS3Path` no need be called in `AdjustPath`.